ledger/ledger-api-tests/infrastructure/src/main/scala/com/daml/ledger/api/testtool/infrastructure/participant/TimeoutParticipantTestContext.scala

```diff
@@ -0,0 +1,497 @@
+import com.daml.ledger.api.v1.ledger_offset.LedgerOffset
+import com.daml.ledger.api.v1.package_service.{GetPackageResponse, PackageStatus}
+import com.daml.ledger.api.v1.transaction.{Transaction, TransactionTree}
+import com.daml.ledger.api.v1.transaction_service.{
+  GetTransactionByEventIdRequest,
+  GetTransactionByIdRequest,
+  GetTransactionsRequest,
+  GetTransactionsResponse,
+}
+import com.daml.ledger.api.v1.value.{Identifier, Value}
+import com.daml.ledger.client.binding.{Primitive, Template}
+import com.daml.lf.data.Ref.HexString
+import com.daml.timer.Delayed
+import com.google.protobuf.ByteString
+import io.grpc.health.v1.health.HealthCheckResponse
+import io.grpc.stub.StreamObserver
+
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{ExecutionContext, Future}
+
+class TimeoutParticipantTestContext(timeoutScaleFactor: Double, delegate: ParticipantTestContext)
+    extends ParticipantTestContext {
+
+  override val ledgerId: String = delegate.ledgerId
+  override val applicationId: String = delegate.applicationId
+  override val endpointId: String = delegate.endpointId
+  override private[participant] def services = delegate.services
+  override private[participant] implicit val ec: ExecutionContext = delegate.ec
+  override def ledgerEndpoint: Endpoint = delegate.ledgerEndpoint
+  override def features: Features = delegate.features
+  override def referenceOffset: LedgerOffset = delegate.referenceOffset
+  override def nextKeyId: () => String = delegate.nextKeyId
+  override def nextUserId: () => String = delegate.nextUserId
+  override def delayMechanism: DelayMechanism = delegate.delayMechanism
+
+  override def currentEnd(): Future[LedgerOffset] =
+    withTimeout("Get current end", delegate.currentEnd())
+  override def currentEnd(overrideLedgerId: String): Future[LedgerOffset] = withTimeout(
+    s"Get current end for ledger id $overrideLedgerId",
+    delegate.currentEnd(overrideLedgerId),
+  )
+  override def offsetBeyondLedgerEnd(): Future[LedgerOffset] = withTimeout(
+    "Offset beyond ledger end",
+    delegate.offsetBeyondLedgerEnd(),
+  )
+  override def time(): Future[Instant] = withTimeout("Get time", delegate.time())
+  override def setTime(currentTime: Instant, newTime: Instant): Future[Unit] = withTimeout(
+    "Set time",
+    delegate.setTime(currentTime, newTime),
+  )
+  override def listKnownPackages(): Future[Seq[PackageDetails]] = withTimeout(
+    "List known packages",
+    delegate.listKnownPackages(),
+  )
+  override def uploadDarRequest(bytes: ByteString): UploadDarFileRequest =
+    delegate.uploadDarRequest(bytes)
+  override def uploadDarFile(request: UploadDarFileRequest): Future[Unit] = withTimeout(
+    s"Upload dar file ${request.submissionId}",
+    delegate.uploadDarFile(request),
+  )
+  override def participantId(): Future[String] =
+    withTimeout("Get participant id", delegate.participantId())
```

Comment:
I'm wondering if a method like this requires a timeout.

Reply:
The main point is that we don't know. But as it's a future, best to add it

Reply:
Make sense